### DISCUSSION TITLE
Error on NT 4

### DISCUSSION THREAD

**Discussion by nick99nack** [wqweto/VbAsyncSocket#32] [Q&A] [answered]
Hello,

I was using the following vbscript to test SSL connections: 

```
Const WinHttpRequestOption_SslErrorIgnoreFlags = 4
Const SslErrorFlag_UnknownCA = 256

With CreateObject("HttpRequest.cHttpRequest")
    .Open_ "GET", "https://www.howsmyssl.com/a/check"
    .Option_(WinHttpRequestOption_SslErrorIgnoreFlags) = SslErrorFlag_UnknownCA
    .Send
    WScript.echo Replace(.ResponseText, ",", vbCrLf)
End With
```

On Windows 2000 and up, it connects fine and returns the proper response from the site. On NT 4, it errors with "Automation Error: The object invoked has disconnected from its clients. Source: cHttpRequest.Send"

Any ideas as to why this would be? 

Edit: This only happens if I try to connect using https, it works correctly if I change the URL to a plain http site.

Windows NT 4.0 SP6a, CScript version 5.6

**Comment by wqweto** [comment]
Do you have Service Pack 6 installed? Try running `winver.exe`.

![image](https://user-images.githubusercontent.com/987963/225242274-64170d2b-702e-4379-9e10-88dd54e5238f.png)

Edit: No I see you have `Windows NT 4.0 SP6a, CScript version 5.6`

This `aaa.vbs` works here on NT4 + SP6 under VirtualBox
```
Const WinHttpRequestOption_SslErrorIgnoreFlags = 4
Const SslErrorFlag_UnknownCA = 256

With CreateObject("HttpRequest.cHttpRequest")
    .Open_ "GET", "https://www.howsmyssl.com/a/check"
    .Option_(WinHttpRequestOption_SslErrorIgnoreFlags) = SslErrorFlag_UnknownCA
    .Send
    WScript.echo Replace(.ResponseText, ",", vbCrLf)
End With
```
![image](https://user-images.githubusercontent.com/987963/225242362-108585aa-8fca-4f5c-a833-fffec9091c40.png)

**Comment by nick99nack** [comment]
Upon doing some further testing, I believe it may be a CPU issue. On Pentium II and III systems, it errors. On Pentium 4, Pentium M and up, it works fine.

**Reply by wqweto** [reply]
Yes, this must be it. The C/C++ thunks are compiled with SSE2 by default enabled.

I’ll have to precise claims of the repo working on NT4 w/ P4 CPU at least.

**Comment by wqweto** [comment, marked as answer]
@nick99nack Latest [release 1.0.4](https://github.com/wqweto/VbAsyncSocket/releases/tag/HRR-1.0.4) has thunks recompiled with `/arch:IA32` option so everyhting must work as expected on Pentium 2 (tested w/ 86Box emulator) and probably earlier (not tested).

**Reply by nick99nack** [reply]
Awesome work. Just tested it and it works great. Thanks!!
